TestLoopbackHostPortIPv6 fails on non IPv6 environment

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


**What happened**:

TestLoopbackHostPortIPv6 fails when run in the machine where there is no IPv6 loopback device configured, with following error:

```
root@43386c5e349b:~/kubernetes# make check WHAT=./vendor/k8s.io/apiserver/pkg/server GOFLAGS=-v KUBE_TEST_ARGS='-run ^TestLoopbackHostPortIPv6 $'
+++ [0831 17:45:32] Running tests without code coverage
=== RUN   TestLoopbackHostPortIPv6
    config_selfclient_test.go:63: no ipv6 loopback interface
--- FAIL: TestLoopbackHostPortIPv6 (0.00s)
FAIL
FAIL	k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server	0.051s
FAIL
make: *** [Makefile:185: check] Error 1
root@43386c5e349b:~/kubernetes#
```

Especially running in the containers where IPv6 is not really configured for the runtime

**What you expected to happen**:

Unit tests shouldn't really depend on the environment it is running,

**How to reproduce it (as minimally and precisely as possible)**:

```
make check WHAT=./vendor/k8s.io/apiserver/pkg/server GOFLAGS=-v KUBE_TEST_ARGS='-run ^TestLoopbackHostPortIPv6 $'
```
**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): master

--- a/staging/src/k8s.io/apiserver/pkg/server/config_selfclient_test.go
+++ b/staging/src/k8s.io/apiserver/pkg/server/config_selfclient_test.go
@@ -60,7 +60,7 @@ func TestLoopbackHostPortIPv6(t *testing.T) {
 		t.Fatalf("fail to enumerate network interface, %s", err)
 	}
 	if !ipv6 {
-		t.Fatalf("no ipv6 loopback interface")
+		t.Skip("no ipv6 loopback interface")
 	}
 
 	host, port, err := LoopbackHostPort("[ff06:0:0:0:0:0:0:c3]:443")

